### ISSUE TITLE
Launching second USB VM for additional Thunderbolt 3/USB busses will freeze the system

### ISSUE THREAD

**Issue by noseshimself** [QubesOS/qubes-issues#4462]
### Qubes OS version:

R4.0

### Affected component(s): 

---

### Steps to reproduce the behavior:

Install Qubes on a current generation Lenovo Thinkpad or similar system with Thunderbolt 3 ports.
Insert a USB device into one of the Thunderbolt/USB ports before booting.
You will find additional PCI devices; one of them will be
```3a:00.0 USB controller: Intel Corporation Device 15ec (rev 06)```.
Clone sys-usb (carefully not autostarting it for obvious reasons) and assign this device to it.
qvm-start sys-usb-2

### Expected behavior:

A happily running system. Maybe even getting access to the device(s) connected to this bus.

### Actual behavior:

System freezes immediately.

### General notes:

---

**Comment by marmarek** [comment]
Can you show `lspci -t` fragment containing this device? What other new devices appear? What happen if you assign all of them to new sys-usb-2?
Do you see any relevant messages in /var/log/xen/console/hypervisor.log after reboot (probably didn't managed to hit the disk, but worth a try).

**Comment by noseshimself** [comment]
One more way to cause a spontaneous reboot: (sys-usb-2 is not around, nothing is connected to the new devices showing up.) Connect an external display using a USB-C cable instead of a DP cable, go to Dom0's display settings and modify something. Turns the machine off, too. Restarting it with the monitor still connected results in a boot loop at the point where X is brought up. Still nothing in hypervisor.log.

**Comment by noseshimself** [comment]
New BIOS, more strangeness:

I updated my system using the currently latest firmware package (1.14 - N2CUR07W) hoping it would solve the (Intel) GPU problems with this system. I do not know what they did but:

If you boot the system with a USB-only device connected to the C-connectors (that includes the new Thunderbolt docking station!) the aforementioned devices come up; connecting something to an empty port (even if it is just a C-to-A adapter with a device connected to it) doesn't show up.

If you connect a USB device to a running machine it will now show up in sys-usb:

[ 8538.125211] usb 2-4: new high-speed USB device number 14 using xhci_hcd
[ 8538.267335] usb 2-4: New USB device found, idVendor=1b1c, idProduct=1a0a
[ 8538.267364] usb 2-4: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[ 8538.267393] usb 2-4: Product: Survivor 3.0
[ 8538.267420] usb 2-4: Manufacturer: Corsair
[ 8538.293019] usb-storage 2-4:1.0: USB Mass Storage device detected
[ 8538.293424] scsi host2: usb-storage 2-4:1.0
[ 8538.293494] usbcore: registered new interface driver usb-storage
[ 8538.295045] usbcore: registered new interface driver uas
[ 8539.297012] scsi 2:0:0:0: Direct-Access     Corsair  Survivor 3.0     000A PQ: 0 ANSI: 6
[ 8539.297616] sd 2:0:0:0: Attached scsi generic sg0 type 0

I'm a bit clueless now. It is clear that the devices showing up if something is connected at boot need to be dealt with somehow (especially at runtime or Ques will not be abe to deal with Thunderbolt within the next decade) and USB detection during Qubes setting up sys-usb is causing the installation process to fail (should I report that separately, too?). But the BIOS and its handling of USB devices as part of the "USB"-C ports seems to be a seriously moving target so any reports here might change overnight.

**Comment by DemiMarie** [comment]
This sounds like a kernel panic in dom0.  Is there a way to get the panic message?

**Comment by str4d** [comment]
I've encountered this issue as well:

- Qubes 4.0.1
- Lenovo Thinkpad P1 (with BIOS 1.18 - this is my second system board, after having bricked the first one in mid-December on the 1.15 BIOS due to a bug with enabling discrete graphics)
- No external USB devices plugged into any port

It has the following USB controllers:
```
00:14.0 USB controller [0c03]: Intel Corporation Device [8086:a36d]
3a:00.0 USB controller [0c03]: Intel Corporation Device [8086:15ec]
```

In my case, during post-install configuration, if I try and create a sys-usb qube then eventually during configuration the laptop locks up (modulo various other bugs I have to work around).

I set up a working Qubes install without sys-usb, then manually created the sys-usb qube, explicitly disabled it from starting on startup (for the obvious reason), and did some testing:

- If I start the sys-usb qube as-is (with both controllers attached), the laptop locks up after maybe 5-10 seconds.
- If I run `qvm-pci detach sys-usb dom0:3a_00.0` first, and then start the sys-usb qube, it starts up as I would normally expect. Internal USB devices attached to it (e.g. the camera) work fine, and I can plug in a USB flash drive into both USB A ports and it appears in the devices menu.

So I'm +1 on this being a Thunderbolt / USB-C issue.
